uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/SearchRESTController.java

```diff
@@ -0,0 +1,162 @@
+ * - People
+ * - Portlet names and descriptions
+ *
+ * JSON results example:
+ *
+ * {"people" : [...],
+ *  "portlets" : [...] }
+ *
+ * @since 5.0
+ **/
+@Controller
+@RequestMapping("/search")
+public final class SearchRESTController {
+
+    private static final Logger logger = LoggerFactory.getLogger(SearchRESTController.class);
+
+    @Autowired
+    private ObjectMapper jsonMapper;
+
+    @Autowired
+    private IPersonManager personManager;
+
+    private PersonLookupHelperImpl lookupHelper;
+
+    @Autowired(required = true)
+    public void setPersonLookupHelper(PersonLookupHelperImpl lookupHelper) {
+        this.lookupHelper = lookupHelper;
+    }
+
+    @Resource(name = "directoryQueryAttributes")
+    private List<String> directoryQueryAttributes;
+
+    /*
+    public void setDirectoryQueryAttributes(List<String> attributes) {
+        this.directoryQueryAttributes = attributes;
+    }
+    */
+
+    @Autowired
+    private IPortletDefinitionRegistry portletDefinitionRegistry;
+
+    @Autowired
+    private PortletRegistryUtil portletRegistryUtil;
+
+    @RequestMapping(method = RequestMethod.GET)
+    public void search(
+            @RequestParam("q") String query,
+            HttpServletRequest request,
+            HttpServletResponse response)
+            throws IOException {
+
+        logger.error("entering search");
+
+        List<Object> matchingPeople = getMatchingPeople(query, request);
+
+        List<Object> matchingPortlets = getMatchingPortlets(query, request);
+
+        if (matchingPeople.isEmpty() && matchingPortlets.isEmpty()) {
+            logger.error("nothing found");
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+        } else {
+            logger.error("send back json");
+            Map<String,List<Object>> results = new TreeMap<>();
+            results.put("people", matchingPeople);
+            results.put("portlets", matchingPortlets);
+            response.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+            jsonMapper.writeValue(response.getOutputStream(), results);
+        }
+    }
+
+    private List<Object> getMatchingPortlets(String query, HttpServletRequest request) {
+        List<Object> results = new ArrayList<>();
+
+        final List<IPortletDefinition> portlets = portletDefinitionRegistry.getAllPortletDefinitions();
+        for (IPortletDefinition portlet : portlets) {
+            if (portletRegistryUtil.matches(query, portlet)) {
+                String url = portletRegistryUtil.buildPortletUrl(request, portlet);
+                if (url != null) {
+                    results.add(getPortletAttrs(portlet, url));
+                }
+            }
+        }
+        return results;
+    }
+
+    private Map<String, String> getPortletAttrs(IPortletDefinition portlet, String url) {
+        Map<String, String> attrs = new TreeMap<>();
+        /* TODO: move this list to Spring configuration xml */
```

Comment:
:question: is this planned for the current PR or a future one?

Reply:
I removed it. It was a nice idea until I realized the attributes are fixed, unlike user attributes. User attributes vary in interest. Portlet defs do not vary between clients.